blockchain/chain_sync/src/sync.rs

```diff
@@ -1104,16 +1110,19 @@ mod tests {
         task::block_on(async move {
             cs.peer_manager.add_peer(source.clone(), None).await;
             assert_eq!(cs.peer_manager.len().await, 1);
-
-            let return_set = cs.sync_headers_reverse(head, &to).await;
-            assert_eq!(return_set.unwrap().len(), 4);
+            // make blocksync request
+            let return_set = task::spawn(async move { cs.sync_headers_reverse(head, &to).await });
+            task::sleep(Duration::from_secs(1)).await;
```

Comment:
You shouldn't need the sleep now that you're awaiting the receiver channel and then sending the response?